llvm/lib/Target/AArch64/AArch64LoadStoreOptimizer.cpp

```diff
@@ -481,6 +505,16 @@ static unsigned getPreIndexedOpcode(unsigned Opc) {
   }
 }
 
+static unsigned getBaseAddressOpcode(unsigned Opc) {
+  // TODO: Add more index address loads/stores.
+  switch (Opc) {
+  default:
+    llvm_unreachable("Opcode has no base address equivalent!");
```

Comment:
At first glance this doesn't look right to me. Are there any checks to actually prevent us getting here with other opcodes? If not, then I think you may need to fill in the remaining opcodes in the switch statement, similar to `getPostIndexedOpcode`.

Reply:
yes, similar to above getLdStAmountOp，I only support `LDRBBroX` in the first version.

Reply:
I guess the point I'm trying to make here is that usually we add a llvm_unreachable because we actually expect the code to be unreachable. However, I can't see anything that prevents us hitting the default case. For example, I'd expect code to be written in this way:

```
void foo(MI) {
  if (opcode == my_expected_opcode)
    foo2(MI);
}

void foo2() {
  switch (MI.getOpcode()) {
  default:
    llvm_unreachable("Unexpected opcode");
  case my_expected_opcode:
    do_something();
    break;
  }
}
```

My question was - do you have anything to explicitly prevent us from calling the function with an unexpected opcode?

Reply:
yes, the function `isMergeableIndexLdSt` called by `AArch64LoadStoreOpt::optimizeBlock` will check only the expected opcode will try this fold. (Now only support one opcode `LDRBBroX` )